decidim/decidim
# Fix faker requirement in prod

#### :tophat: What? Why?

@microstudi posted the following comment in  https://github.com/decidim/decidim/issues/11968#issuecomment-1816117461

    @andreslucena @alecslupu is it possible that this PR has made mandatory to have the gem "faker" installed in production?
I've seen that I need to include it now whereas it wasn't necessary before.

Maybe is because of these lines?

https://github.com/decidim/decidim/blob/92e5aa1d74435bc460106922fa41455511a1f065/decidim-participatory_processes/lib/decidim/participatory_processes/participatory_space.rb#L3

https://github.com/decidim/decidim/blob/92e5aa1d74435bc460106922fa41455511a1f065/decidim-participatory_processes/lib/decidim/participatory_processes/seeds.rb#L3

Looks like the gem "fake" is required always no matter the env.

Got this error if not installed:

```
> [builder 24/27] RUN RAILS_ENV=production     SECRET_KEY_BASE=dummy     RAILS_MASTER_KEY=dummy     DB_ADAPTER=nulldb     bundle exec rails assets:precompile:                                
10.49 rails aborted!                                                                                                                                                                           
10.49 LoadError: cannot load such file -- faker                                                                                                                                                
10.49 /usr/local/bundle/gems/bootsnap-1.17.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:29:in `require'                                                                           
10.49 /usr/local/bundle/gems/activesupport-6.1.7.6/lib/active_support/dependencies.rb:332:in `block in require'                                                                                
10.49 /usr/local/bundle/gems/activesupport-6.1.7.6/lib/active_support/dependencies.rb:299:in `load_dependency'
10.49 /usr/local/bundle/gems/activesupport-6.1.7.6/lib/active_support/dependencies.rb:332:in `require'
10.49 /usr/local/bundle/bundler/gems/decidim-489a76446b40/decidim-participatory_processes/lib/decidim/participatory_processes/seeds.rb:3:in `<main>'
10.49 /usr/local/bundle/gems/bootsnap-1.17.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
10.49 /usr/local/bundle/gems/bootsnap-1.17.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
10.49 /usr/local/bundle/gems/activesupport-6.1.7.6/lib/active_support/dependencies.rb:332:in `block in require'
10.49 /usr/local/bundle/gems/activesupport-6.1.7.6/lib/active_support/dependencies.rb:299:in `load_dependency'
10.49 /usr/local/bundle/gems/activesupport-6.1.7.6/lib/active_support/dependencies.rb:332:in `require'
10.49 /usr/local/bundle/bundler/gems/decidim-489a76446b40/decidim-participatory_processes/lib/decidim/participatory_processes/participatory_space.rb:3:in `<main>'
10.49 /usr/local/bundle/gems/bootsnap-1.17.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
10.49 /usr/local/bundle/gems/bootsnap-1.17.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
```
            

#### :pushpin: Related Issues
*Link your PR to an issue*
- Related to #11968

#### Testing
1. Create the development application 
2. Comment Gemfile and run `bundle clean --force` ( this will clean all the gems in your system !!!)
3. run `BUNDLE_WITHOUT="development:test" bundle install`
4. run `RAILS_ENV=production BUNDLE_WITHOUT="development:test" bundle exec rails assets:precompile`
5. See error 
6. Apply patch 
7. repeat 3 + 4 
8. See there is no error 

:hearts: Thank you!
